crates/ruff_server/src/server/api.rs

```diff
@@ -84,6 +84,15 @@ pub(super) fn notification<'a>(notif: server::Notification) -> Task<'a> {
         }
         notification::DidClose::METHOD => local_notification_task::<notification::DidClose>(notif),
         notification::DidOpen::METHOD => local_notification_task::<notification::DidOpen>(notif),
+        notification::DidOpenNotebook::METHOD => {
+            local_notification_task::<notification::DidOpenNotebook>(notif)
+        }
+        notification::DidChangeNotebook::METHOD => {
+            local_notification_task::<notification::DidChangeNotebook>(notif)
+        }
+        notification::DidCloseNotebook::METHOD => {
+            local_notification_task::<notification::DidCloseNotebook>(notif)
+        }
```

Comment:
I might be missing some context here but can you tell me why is the `DidSaveNotebookDocument` not implemented?

Reply:
At the moment, I don't see a need for us to handle `didSave` notifications in general (we don't handle `didSaveTextDocument` either).